projects/ccf-rui/src/app/modules/info/info-button/info-button.component.ts

```diff
@@ -17,17 +19,29 @@ export class InfoButtonComponent {
    * Creates an instance of info button component.
    *
    * @param dialog Reference to the dialog creation service.
+   * @param infoButtonService Reference to the info button service
    */
-  constructor(private readonly dialog: MatDialog) { }
+  constructor(private readonly dialog: MatDialog, private infoButtonService: InfoButtonService) {
+    infoButtonService.markdownContent.subscribe(data => {
```

Comment:
add `ngOnDestroy` and unsubscribe from the observable

Reply:
Should the unsubscribe be when the modal is closed? The button still remains on the UI throughout

Reply:
No, just on destroy of the info-button (not likely to happen, but its a best practice to unsubscribe from all subscriptions when a component is destroyed).